docs/config_file.md

```diff
@@ -0,0 +1,58 @@
+# Configuration File
+
+There are a lot of CLI options available for `idf-build-apps`. These configuration items provide usage flexibility, but also make the CLI command too long to read.
+
+Now with configuration file support, we could easily declare the configuration items, in a more maintainable way.
+
+## Quickstart
+
+```toml
+[default]
+paths = [
+    "components",
+    "examples",
+]
+target = "esp32"
+recursive = true
+
+# config rules
+config = [
+    "sdkconfig.*=",
+    "=default",
+]
+
+# build related options
+build_dir = "build_@t_@w"
+```
+
+Run `idf-build-apps build` with this configuration file is equivalent to the CLI command `idf-build-app build --paths components examples --target esp32 --recursive --config "sdkconfig.*=" --config "=default" --build-dir "build_@t_@w"` without it.
+
+The configuration file is a [TOML][toml] file. [TOML][toml] is a configuration file language that is widely used in Python. The native support in the standard library for [TOML][toml] is added in python 3.11.
+
+The configuration items declared in the configuration file are almost the same as the CLI options. Only need to mention a few key points:
+
+1. The name of the configuration item is different. Please replace all dashes `-` to underscores `_`.
+2. [TOML][toml] has native types. Use the same type as the CLI options ones.
+
+## Usage
+
+By default, the `.idf-build-apps.toml` file under your current path will be considered as the root configuration file, and the table `default` will be considered as the default table of the toml file. ("[table][toml-table]" is a term in toml file)
+
+To pass custom values, you may call `--config-file <config file path> --config-table <table name>`.
+
+It's very handy when you're setting up different configurations in CI and in your local dev environment.
```

Comment:
FWIW, if you are setting up a configuration for your local dev environment, you would probably just create a different TOML file, to avoid accidentally committing local changes to the config file checked into Git...

Reply:
:man_facepalming: I agree. it's a bad use case.